flake8_fine_pytest/watchers/unique_test_names.py

```diff
@@ -0,0 +1,35 @@
+import ast
+from typing import Set
+
+from flake8_fine_pytest.watchers.base import BaseWatcher
+
+
+class UniqueTestNamesWatcher(BaseWatcher):
+    error_template = 'FP009 Duplicate name test case ({})'
+
+    def run(self) -> None:
+        testcases_names: Set[str] = set()
+        duplicates: Set[str] = set()
+
+        if not self._is_test_file(self.filename):
+            return
+
+        for node in ast.walk(self.tree):
+            if not self._is_properly_node(node):
+                continue
+
+            node_name = node.name  # type: ignore
+
+            if node_name not in testcases_names:
```

Comment:
Why is this construction needed? I think the algorithm can be implemented without `testcases_names` set.

Reply:
I couldn't without testcases_names, but I was able to simplify the algorithm  
https://github.com/best-doctor/flake8-fine-pytest/pull/7/commits/1ac73269540ec98da1dd65bd55b9513b40915a2d